diff --git a/images/build/debian-base/Makefile b/images/build/debian-base/Makefile
index db1576d0..f82b2e82 100755
--- a/images/build/debian-base/Makefile
+++ b/images/build/debian-base/Makefile
@@ -24,10 +24,11 @@ CONFIG ?= bookworm
 
 TAR_FILE ?= rootfs.tar
 ARCH ?= amd64
-ALL_ARCH = amd64 arm arm64 ppc64le s390x
+ALL_ARCH = amd64 arm arm64 ppc64le s390x loong64
 
-QEMUVERSION=7.2.0-1
+QEMUVERSION ?= 7.2.0-1
 QEMUIMAGE ?= multiarch/qemu-user-static
+QEMUFLAGS ?= --reset -p yes
 
 # This option is for running docker manifest command
 export DOCKER_CLI_EXPERIMENTAL := enabled
@@ -52,6 +53,10 @@ ifeq ($(ARCH),s390x)
 	BASEIMAGE?=s390x/debian:$(CONFIG)-slim
 	QEMUARCH=s390x
 endif
+ifeq ($(ARCH),loong64)
+	BASEIMAGE?=loong64/debian:$(CONFIG)-slim
+	QEMUARCH=loongarch64
+endif
 
 sub-build-%:
 	$(MAKE) ARCH=$* build
@@ -75,7 +80,7 @@ build: clean
 	umask 0022
 
 	# Enable execution of multi-architecture containers
-	docker run --rm --privileged $(QEMUIMAGE):$(QEMUVERSION) --reset -p yes
+	docker run --rm --privileged $(QEMUIMAGE):$(QEMUVERSION) $(QEMUFLAGS)
 	docker buildx version
 	BUILDER=$(shell docker buildx create --use)
 
@@ -94,16 +99,12 @@ build: clean
 		--load \
 		--platform linux/$(ARCH) \
 		-t $(IMAGE)-$(ARCH):$(IMAGE_VERSION) \
-		-t $(IMAGE)-$(ARCH):$(TAG)-$(CONFIG) \
-		-t $(IMAGE)-$(ARCH):latest-$(CONFIG) \
 		$(CONFIG)
 	docker buildx rm $$BUILDER
 	rm $(CONFIG)/rootfs.tar
 
 push: build
 	docker push $(IMAGE)-$(ARCH):$(IMAGE_VERSION)
-	docker push $(IMAGE)-$(ARCH):$(TAG)-$(CONFIG)
-	docker push $(IMAGE)-$(ARCH):latest-$(CONFIG)
 
 clean:
 	docker rmi -f $(IMAGE)-$(ARCH):$(IMAGE_VERSION) || true
