diff --git a/images/Makefile.common-image b/images/Makefile.common-image
index 9b08ccaf..b58a7b61 100644
--- a/images/Makefile.common-image
+++ b/images/Makefile.common-image
@@ -19,6 +19,8 @@ IMAGE = $(REGISTRY)/$(IMGNAME)
 
 TAG ?= $(shell git describe --tags --always --dirty)
 
+SERIAL ?= 0
+
 # TODO: Uncomment once all images using this Makefile can be built on all
 #       supported Kubernetes server platforms.
 #PLATFORMS ?= linux/amd64 linux/arm linux/arm64 linux/ppc64le linux/s390x
@@ -45,8 +47,9 @@ container: check-env init-docker-buildx
 			--progress plain \
 			--platform $${platform} \
 			--tag $(IMAGE)-$${platform##*/}:$(IMAGE_VERSION) \
-			--tag $(IMAGE)-$${platform##*/}:$(TAG)-$(CONFIG) \
-			--tag $(IMAGE)-$${platform##*/}:latest-$(CONFIG) \
+			--tag $(IMAGE)-$${platform##*/}:$(IMAGE_VERSION)-$(SERIAL) \
+            --tag $(IMAGE):$(IMAGE_VERSION) \
+            --tag $(IMAGE):$(IMAGE_VERSION)-$(SERIAL) \
 			$(BUILD_ARGS) \
 			.; \
 	done
@@ -57,8 +60,9 @@ push: container
 	@for platform in $(PLATFORMS); do \
 		echo "Pushing tags for $${platform} platform"; \
 		docker push $(IMAGE)-$${platform##*/}:$(IMAGE_VERSION); \
-		docker push $(IMAGE)-$${platform##*/}:$(TAG)-$(CONFIG); \
-		docker push $(IMAGE)-$${platform##*/}:latest-$(CONFIG); \
+		docker push $(IMAGE)-$${platform##*/}:$(IMAGE_VERSION)-$(SERIAL); \
+        docker push $(IMAGE):$(IMAGE_VERSION); \
+        docker push $(IMAGE):$(IMAGE_VERSION)-$(SERIAL); \
 	done
 
 .PHONY: manifest
diff --git a/images/build/debian-base/Makefile b/images/build/debian-base/Makefile
index db1576d0..cb25e4a2 100755
--- a/images/build/debian-base/Makefile
+++ b/images/build/debian-base/Makefile
@@ -22,12 +22,14 @@ TAG ?= $(shell git describe --tags --always --dirty)
 IMAGE_VERSION ?= bookworm-v1.0.7
 CONFIG ?= bookworm
 
+SERIAL ?= 0
 TAR_FILE ?= rootfs.tar
 ARCH ?= amd64
-ALL_ARCH = amd64 arm arm64 ppc64le s390x
+ALL_ARCH = amd64 arm arm64 ppc64le s390x loong64
 
-QEMUVERSION=7.2.0-1
+QEMUVERSION ?= 7.2.0-1
 QEMUIMAGE ?= multiarch/qemu-user-static
+QEMUFLAGS ?= --reset -p yes
 
 # This option is for running docker manifest command
 export DOCKER_CLI_EXPERIMENTAL := enabled
@@ -52,6 +54,10 @@ ifeq ($(ARCH),s390x)
 	BASEIMAGE?=s390x/debian:$(CONFIG)-slim
 	QEMUARCH=s390x
 endif
+ifeq ($(ARCH),loong64)
+	BASEIMAGE?=loong64/debian:$(CONFIG)-slim
+	QEMUARCH=loongarch64
+endif
 
 sub-build-%:
 	$(MAKE) ARCH=$* build
@@ -75,7 +81,7 @@ build: clean
 	umask 0022
 
 	# Enable execution of multi-architecture containers
-	docker run --rm --privileged $(QEMUIMAGE):$(QEMUVERSION) --reset -p yes
+	docker run --rm --privileged $(QEMUIMAGE):$(QEMUVERSION) $(QEMUFLAGS)
 	docker buildx version
 	BUILDER=$(shell docker buildx create --use)
 
@@ -94,16 +100,14 @@ build: clean
 		--load \
 		--platform linux/$(ARCH) \
 		-t $(IMAGE)-$(ARCH):$(IMAGE_VERSION) \
-		-t $(IMAGE)-$(ARCH):$(TAG)-$(CONFIG) \
-		-t $(IMAGE)-$(ARCH):latest-$(CONFIG) \
+		-t $(IMAGE)-$(ARCH):$(IMAGE_VERSION)-$(SERIAL) \
 		$(CONFIG)
 	docker buildx rm $$BUILDER
 	rm $(CONFIG)/rootfs.tar
 
 push: build
 	docker push $(IMAGE)-$(ARCH):$(IMAGE_VERSION)
-	docker push $(IMAGE)-$(ARCH):$(TAG)-$(CONFIG)
-	docker push $(IMAGE)-$(ARCH):latest-$(CONFIG)
+	docker push $(IMAGE)-$(ARCH):$(IMAGE_VERSION)-$(SERIAL)
 
 clean:
 	docker rmi -f $(IMAGE)-$(ARCH):$(IMAGE_VERSION) || true
diff --git a/images/build/distroless-iptables/Makefile b/images/build/distroless-iptables/Makefile
index b7517313..ffd2ffd0 100644
--- a/images/build/distroless-iptables/Makefile
+++ b/images/build/distroless-iptables/Makefile
@@ -23,6 +23,8 @@ CONFIG ?= distroless-bookworm
 BASEIMAGE ?= debian:bookworm-slim
 GORUNNER_VERSION ?= v2.4.0-go1.25.9-bookworm.0
 
+SERIAL ?= 0
+
 ARCH?=amd64
 ALL_ARCH = amd64 arm arm64 ppc64le s390x
 
@@ -30,8 +32,9 @@ BASE_REGISTRY?=registry.k8s.io/build-image
 
 GORUNNERIMAGE?=$(BASE_REGISTRY)/go-runner:$(GORUNNER_VERSION)
 
-QEMUVERSION=7.2.0-1
+QEMUVERSION ?= 7.2.0-1
 QEMUIMAGE ?= multiarch/qemu-user-static
+QEMUFLAGS ?= --reset -p yes
 
 ifneq ($(ARCH), amd64)
 	SKIP_WRAPPER_CHECK=--no-sanity-check
@@ -47,7 +50,7 @@ build:
 	umask 0022
 
 	# Enable execution of multi-architecture containers
-	docker run --rm --privileged $(QEMUIMAGE):$(QEMUVERSION) --reset -p yes
+	docker run --rm --privileged $(QEMUIMAGE):$(QEMUVERSION) $(QEMUFLAGS)
 	docker buildx version
 	BUILDER=$(shell docker buildx create --use)
 	docker buildx build \
@@ -55,8 +58,7 @@ build:
 		--load \
 		--platform linux/$(ARCH) \
 		-t $(IMAGE)-$(ARCH):$(IMAGE_VERSION) \
-		-t $(IMAGE)-$(ARCH):$(TAG)-$(CONFIG) \
-		-t $(IMAGE)-$(ARCH):latest-$(CONFIG) \
+		-t $(IMAGE)-$(ARCH):$(IMAGE_VERSION)-$(SERIAL) \
 		--build-arg=BASEIMAGE=$(BASEIMAGE) \
 		--build-arg=GORUNNERIMAGE=$(GORUNNERIMAGE) \
 		--build-arg=SKIP_WRAPPER_CHECK=$(SKIP_WRAPPER_CHECK) \
@@ -65,8 +67,7 @@ build:
 
 push: build
 	docker push $(IMAGE)-$(ARCH):$(IMAGE_VERSION)
-	docker push $(IMAGE)-$(ARCH):$(TAG)-$(CONFIG)
-	docker push $(IMAGE)-$(ARCH):latest-$(CONFIG)
+	docker push $(IMAGE)-$(ARCH):$(IMAGE_VERSION)-$(SERIAL)
 
 sub-build-%:
 	$(MAKE) ARCH=$* build
diff --git a/images/build/distroless-iptables/distroless-bookworm/Dockerfile b/images/build/distroless-iptables/distroless-bookworm/Dockerfile
index 8e263ac0..f9ed28b2 100644
--- a/images/build/distroless-iptables/distroless-bookworm/Dockerfile
+++ b/images/build/distroless-iptables/distroless-bookworm/Dockerfile
@@ -51,7 +51,7 @@ FROM ${GORUNNERIMAGE} as gorunner
 
 # We need to use distroless:base here as tzdata, glibc and some other base packages
 # are required
-FROM gcr.io/distroless/base-debian12 as intermediate
+FROM ghcr.io/loong64/debian:trixie-slim as intermediate
 
 ARG SKIP_WRAPPER_CHECK
 
diff --git a/images/build/setcap/Makefile b/images/build/setcap/Makefile
index 6bd7ce6c..619f1ecc 100644
--- a/images/build/setcap/Makefile
+++ b/images/build/setcap/Makefile
@@ -22,6 +22,7 @@ IMAGE_VERSION ?= bookworm-v1.0.7
 CONFIG ?= bookworm
 DEBIAN_BASE_VERSION ?= bookworm-v1.0.7
 
+SERIAL ?= 0
 ARCH?=amd64
 ALL_ARCH = amd64 arm arm64 ppc64le s390x
 
@@ -29,8 +30,9 @@ BASE_REGISTRY?=registry.k8s.io/build-image
 BASEIMAGE?=$(BASE_REGISTRY)/debian-base-$(ARCH):$(DEBIAN_BASE_VERSION)
 
 # Build args
-QEMUVERSION=7.2.0-1
+QEMUVERSION ?= 7.2.0-1
 QEMUIMAGE ?= multiarch/qemu-user-static
+QEMUFLAGS ?= --reset -p yes
 
 # This option is for running docker manifest command
 export DOCKER_CLI_EXPERIMENTAL := enabled
@@ -40,7 +42,7 @@ build:
 	umask 0022
 
 	# Enable execution of multi-architecture containers
-	docker run --rm --privileged $(QEMUIMAGE):$(QEMUVERSION) --reset -p yes
+	docker run --rm --privileged $(QEMUIMAGE):$(QEMUVERSION) $(QEMUFLAGS)
 	docker buildx version
 	BUILDER=$(shell docker buildx create --use)
 	docker buildx build \
@@ -48,16 +50,14 @@ build:
 		--load \
 		--platform linux/$(ARCH) \
 		-t $(IMAGE)-$(ARCH):$(IMAGE_VERSION) \
-		-t $(IMAGE)-$(ARCH):$(TAG)-$(CONFIG) \
-		-t $(IMAGE)-$(ARCH):latest-$(CONFIG) \
+		-t $(IMAGE)-$(ARCH):$(IMAGE_VERSION)-$(SERIAL) \
 		--build-arg=BASEIMAGE=$(BASEIMAGE) \
 		.
 	docker buildx rm $$BUILDER
 
 push: build
 	docker push $(IMAGE)-$(ARCH):$(IMAGE_VERSION)
-	docker push $(IMAGE)-$(ARCH):$(TAG)-$(CONFIG)
-	docker push $(IMAGE)-$(ARCH):latest-$(CONFIG)
+	docker push $(IMAGE)-$(ARCH):$(IMAGE_VERSION)-$(SERIAL)
 
 sub-build-%:
 	$(MAKE) ARCH=$* build
